homeassistant/components/wmspro/cover.py

```diff
@@ -0,0 +1,78 @@
+"""Support for covers connected with WMS WebControl pro."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any
+
+from wmspro.const import (
+    WMS_WebControl_pro_API_actionDescription,
+    WMS_WebControl_pro_API_actionType,
+)
+
+from homeassistant.components.cover import ATTR_POSITION, CoverDeviceClass, CoverEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .generic_entity import WebControlProGenericEntity
+
+SCAN_INTERVAL = timedelta(seconds=5)
```

Comment:
Can you elaborate on what a Destination is? Will you add more future platforms? What happens during a refresh?

Reply:
A Destination is basically a device control channel. A device can have multiple channels (motor, light, ...) and a channel can have multiple actions.

Yes, I have fully working support for scenes and lights ready locally, as described in the PR description. Additional requests can come from the community based on diagnostics output, as I currently only have an awning with lighting. So yes, I plan to add more platforms in the future and maintain this integration.

A refresh on the hub calls the getConfiguration API endpoint to retrieve all destinations (aka devices), rooms and scenes. A refresh on a destination retreives the actions of the device. Interestingly an action is both something you perform as well as the current state of the action related component, e.g. the awning drive has a position and can be moved. So, refresh here retrieves the list of actions available and their current state.

Reply:
More like, can one destination spawn more entities? How does the update work technically?

Reply:
A destination may spawn more than one entity if it has actions that would make sense to represent with different platform due to their capabilities, e.g. an awning with integrated lightning or multiple motors (e.g. to control a valance). At the moment I am lacking such devices, but in theory the WebControl API could expose such a destination with such a variety of actions/capabilities supported.

You can see what happens after the REST API call here:
https://github.com/mback2k/pywmspro/blob/e4cfdb16201b28068a97b52db49e10520bad5e8f/wmspro/destination.py#L59-L79

Reply:
Would it make sense to have the polling centralized using a coordinator? This way its easier to add more platforms to the same destination in the future

Reply:
Since each destination needs to polled with a dedicated API call, I don't see the need to do this now. It could be a future improvement if such devices really appear in the wild. For example my awning exposes its light as a dedicated destination.

Reply:
But you could create a coordinator per device. This would allow us to setup the devices in `__init__.py` and just have the platforms add them whenever possible

Reply:
Can you point me at an example please? I couldn't figure out coordinators based on the available documentation and the current approach way more straightforward. What advantage does a coordinator per device have over the simple async_update per entity logic?

Reply:
airgradient for example. I think the main benefit is that it allows you to easily attach more platforms. Usually I recommend it, but in this case I think it's kinda equal.

Reply:
But let's skip it for now then, was just checking if it was relevant or not

Reply:
Okay, thanks, but I will keep this in my head for later platform and device additions, especially if the number of supported destinations gets out of hand. :)